data_processor_node:
  ros__parameters:
    # processing configuration
    processing:
      # lidar configuration
      lidar:
        # z-offset of the LiDAR w.r.t. ground level
        height_offset: 1.73
        # filter bounding box corresponding to the contour of the vehicle w.r.t. lidar frame
        bounding_box:
          # bounding box is specified in counterclockwise direction, from front right corner
          # the below are coordinate pairs [x, y]
          - [2.79, 0.8]
          - [2.79, -0.8]
          - [-1.62, -0.8]
          - [-1.62, 0.8]
        # segmentation configuration
        segmentation:
          # algorithm to be used for segmentation
          algorithm: "ransac"
          # parameters used by RANSAC
          ransac:
            orthogonal_distance_threshold: 0.1
            number_of_iterations: 150
            thread_count: 8
    # visualization configuration
    visualization:
      lidar:
        segmentation:
          enable: true
          topic: "lidar/segmentation"
        clustering:
          enable: true
          topic: "lidar/clustering"
        polygonization:
          enable: true
          topic: "lidar/polygonization"
